src/ssl.c

```diff
@@ -44405,13 +44405,35 @@ int wolfSSL_CTX_set1_curves_list(WOLFSSL_CTX* ctx, const char* names)
                                           (XSTRNCMP(name, "P-521", len) == 0)) {
             curve = WOLFSSL_ECC_SECP521R1;
         }
-        else if (XSTRNCMP(name, "X25519", len) == 0)
+        else if (XSTRNCMP(name, "X25519", len) == 0) {
             curve = WOLFSSL_ECC_X25519;
-        else if ((curve = wc_ecc_get_curve_id_from_name(name)) < 0)
+        }
+        else {
+            int ret = wc_ecc_get_curve_id_from_name(name);
```

Comment:
@JacobBarthelmeh : It looks like the `curve_id` returned here is the `ecc.h` `enum ecc_curve_id`. This does not match the expected parameter for `wolfSSL_CTX_UseSupportedCurve` or the logic for `disabledCurves` in `TLSX_ValidateSupportedCurves`. Those expect the `WOLFSSL_ECC_*` naming in ssl.h "Named Groups" enum ~ line 2897.

Reply:
added a commit to do conversion of name to a value that wolfSSL_CTX_UseSupportedCurve can understand